### PR TITLE
Initial hubot Cloudant scripts

### PR DESCRIPTION
**Pull request by JonHoughton**
Initial commit.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,89 @@
+[![Build Status](https://travis-ci.org/ibm-cloud-solutions/hubot-ibmcloud-cloudant.svg?branch=master)](https://travis-ci.org/ibm-cloud-solutions/hubot-ibmcloud-cloudant)
+[![Coverage Status](https://coveralls.io/repos/github/ibm-cloud-solutions/hubot-ibmcloud-cloudant/badge.svg?branch=cleanup)](https://coveralls.io/github/ibm-cloud-solutions/hubot-ibmcloud-cloudant?branch=cleanup)
```
Comment (chambrid) [inline README.md]: branch should be `master`. This is a bug in the repo you took this from.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    "test": ". test/.env && mocha test",
+    "coverage": ". test/.env && istanbul cover ./node_modules/mocha/bin/_mocha --report lcovonly -- -R spec && cat ./coverage/lcov.info | ./node_modules/coveralls/bin/coveralls.js && rm -rf ./coverage"
+  },
+  "keywords": [
+    "hubot",
+    "hubot-scripts",
+    "bluemix",
+    "cloud",
+    "slack",
+    "cloudant"
```
Comment (chambrid) [inline package.json:27]: we have been including these keywords as well in other repos:  "ibm","ibm cloud"

File: README.md (hunk 1 above)
Comment (JonHoughton) [inline README.md]: Done.

File: package.json (hunk 2 above)
Comment (JonHoughton) [inline package.json:27]: Done.

File: src/nlc/NLC.json [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+				"I'd like to see information for the database",
+				"cloudant info",
+				"I want to see my detailed information for my cloudant database",
+				"Give me database details"
+  			],
+	       "parameters" : [
+			   {
+			       "name": "databasename",
+			       "type": "entity",
+			       "prompt": "OK. What is the name of the database for which you want detailed?"
```
Comment (chambrid) [inline src/nlc/NLC.json]: for which you want details? vs. detailed

File: src/lib/cloudant.js [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+var env = null;
+
+/**
+ * Builds the options used to initialize with Cloudant (account/username/password)
+ * based on the environment variables.
+ *
+ * @return {object} [Object with api object with account/username/password fields]
+ */
+function buildEnv() {
+
```
Comment (chambrid) [inline src/lib/cloudant.js:21]: We should open a follow up issue to pull the env conditionally from VCAP if the service is bound to the container in Bluemix. @ycao56 can help with this.

File: test/cloudant.cognitive.test.js [hunk 5]
```diff
@@ -0,0 +1,352 @@
...
+	defaultLocale: 'en',
+	// Prevent messages file from being overwritten in error conditions (like poor JSON).
+	updateFiles: false
+});
+// At some point we need to toggle this setting based on some user input.
+i18n.setLocale('en');
+
+// Passing arrow functions to mocha is discouraged: https://mochajs.org/#arrow-functions
+// return promises from mocha tests rather than calling done() - http://tobyho.com/2015/12/16/mocha-with-promises/
+describe('Interacting with Cloudant via Slack', function() {
```
Comment (chambrid) [inline test/cloudant.cognitive.test.js]: Perhaps remove slack and highlight RegEx or Natural Language in describe text

**Comment by chambrid** [comment]
A couple comments. Also your CLAs are for app-management. You can grab the word docs from the skeleton project and generate the appropriate PDFs.

File: src/nlc/NLC.json (hunk 3 above)
Comment (JonHoughton) [inline src/nlc/NLC.json]: Done.

File: test/cloudant.cognitive.test.js (hunk 5 above)
Comment (JonHoughton) [inline test/cloudant.cognitive.test.js]: Reworded all references to slack in the three test files.

File: src/lib/cloudant.js (hunk 4 above)
Comment (JonHoughton) [inline src/lib/cloudant.js:21]: Opened issue https://github.com/ibm-cloud-solutions/hubot-ibmcloud-cloudant/issues/3.

**Comment by JonHoughton** [comment]
@chambrid I have made all of the review changes including updating the cla docs.  Thanks for reviewing.  Please let me know if you'd like me to merge this pull request.

**Comment by chambrid** [comment]
👍

**Pull request merged**